### DISCUSSION TITLE
What could non-developer operating staff do in the open source community?

### DISCUSSION THREAD

**Discussion by zhangyangyu** [tisonkun/blog#9] [General]
Recently, we could see some non-developer operating staff in commercial enterprise spamming open source project issues or pull requests. Although they work hard but give opposite bad feeling to contributors. So what's the right thing for non-developer operating staff do in the open source community?

**Comment by tisonkun** [comment]
The first question to answer is what "operation" is necessary from where you stand. I can make some assumptions and try to talk about how staff can work to achieve the goal, but it's firstly your business and you should define "operating" an open-source community.

If you don't know what they can do because you don't know what they're able to do and what goal they are working for, then it's possible that actually, you aren't gonna need it.

**Comment by wu-sheng** [comment]
I agree with @tisonkun, but for more, I think what does `non-developer` mean? In the real open-source community, there are few ppl to identify themselves as non-developer. This is an open-source project, it is software, so, it is definitely relative to software development.

Also, why are you going to involve an open-source community is a key point. If that is your work, then who paid you makes the call, isn't it? If you are going to join because you need it, you are using it, you are building things on top of it, when you are easy to know what is not perfect/good enough in this project/community, you could help from there.

From not developing codes, there are several things people are doing, such as helping a foundation or project get funds, mentoring projects to go through legal and trademark issues (from a layer perspective), organizing Hackathons to help people learn project(but not for statistic, stars, number of contributors, commits). Many could be done if your start point is helping from a deep heart.

If this discussion goes into some vendor or commercial marketing direction, I have no interest to continue. Because I have witnessed many Chinese vendors following this direction, no matter how wrong this is. 

As @tisonkun pinged me on Twitter, this is a short version in my mind.

**Comment by zhangyangyu** [comment]
Thanks for the reply! But I'm afraid my confusion hit

> If this discussion goes into some vendor or commercial marketing direction, I have no interest to continue. Because I have witnessed many Chinese vendors following this direction, no matter how wrong this is.

Many wrong cases have been witnessed, trying to establish connections between projects and contributors, trying to advocate projects or events, eventually spamming Github issues or contributor mails. What's the right way to achieve such goals if I were an operating employee driven by the goals?

**Reply by wu-sheng** [reply]
Between projects connection usually is fine, if some proposals make sense.
But try to get connecting to every contributor, it is not fine at all. Contributors are dedicated to add more to the project, asking more, especially personal information, is very impolite.
People hangout because they feel connected personally, they talked from time to time at on site or on mail list for years. This is just personal behavior. Some are shy and prefer private and in the shadow in project. 
Notice, they do more to join a community, they don't owe a community anything.

**Reply by tisonkun** [reply]
> trying to establish connections between projects and contributors

I have a discussion with @majinghe about such expression that you cannot act as a superior and "connect the project and contributors". Because it's the contributors build the project and the community.

My comments there:

> 开发者和开源项目之间的桥梁是什么比喻？在我的理解是这俩不是对等的，开发者就是项目社区成员，自己跟自己不需要桥梁。社群即包括项目，不存在城堡式的项目是城堡，社群做桥梁和护城河的说法。
> ASF 讲 community over code 其实强调的是代码不会自动演进，只有人才能撑起项目代码的演化、软件的采用和生态的繁荣。重在强调人的重要性，而不是跨过人去考核代码或者把人物化。
> 这种思路我会觉得其实是 community mgr 不是以贡献赢得权威的核心成员，而是一个市场部的管理者，这样他能做的就是按照面向消费者一类的运营手段来认识开源社群。正好是 community over code 想反对的越过人考核指标。把 community over code 解释称没有代码，考核人数和活动的指标，就是完全不知所谓了。
